### ISSUE TITLE
Using this for an text editor?

### ISSUE THREAD

**Issue by knipferrc** [gyscos/Cursive#220]
First: is this a bug report? A suggestion? Or asking for help?
Asking fro help
# Problem description
Im looking to start learning rust and I would like to build a simple text editor, this seems to be the most well maintained TUI library for rust and was wondering if its feasible to use this to make a simple terminal text editor?
If possible code, or pseudo-code to reproduce the problem / present an idea.

# Environment

* Operating system: linux distribution, windows, macOS, ... 
Linux
* Backend used: ncurses (the default one), pancurses, termion, ...
  * If using ncurses on linux, ncurses version? (run `grep "define NCURSES_VERSION " /usr/include/curses.h`)
* Current locale (run `locale` in a terminal).
en_US
* Cursive version (from crates.io, from git, ...)
crates.io

**Comment by matthiasbeyer** [comment]
@knipferrc if you implement said `EditorView` I wouldn't mind a library for this! :smile: 

Also, if you (by any chance) implement a "tab" functionality, I would love to see a library for this as well! 

---

But, as you're learning: No pressure! :smile:

**Comment by knipferrc** [comment]
I would definitely be interested in at least taking a swing at making an `EditorView`, could be a fun learning experience.

**Comment by matthiasbeyer** [comment]
Hell yeah, that is actually a freakin' nice idea! I will try to implemented a `View` using Xi tomorrow, I guess.

**Comment by njskalski** [comment]
https://github.com/njskalski/slime/blob/master/src/slime_text_view.rs

this is so work in progress it's a shame to share at this state. For instance backspace doesn't work :D

**Comment by njskalski** [comment]
@gyscos btw, I would really use some constructive criticism of this code. I'm not fluent with Rust yet. Right now I have a significant delay in drawing (some input work, just run "slime somedirectory"). I will probably profile it next weekend, but if you see some obvious error, let me know. I think it might be the fact I am not using callbacks, but not sure if that's the reason.

**Comment by gyscos** [comment]
I'd love to have a look! But it seems the project is not available - maybe it was removed? Or is not public?

**Comment by njskalski** [comment]
Yes, I have to ask my company for release permission. I already filed for it, waiting for answer.

**Comment by njskalski** [comment]
finally got clearance to share my code. The editor is sly_text_view.rs. It's a bit complicated (separate model from view, multiple cursors, some events bound to configurable input), but can be used as reference. Just make sure you know the license (Apache 2.0).

https://github.com/njskalski/sly-editor
